packages/wells/src/client/api/wells.ts

```diff
@@ -202,6 +233,11 @@ export class Wells extends AssetsAPI {
       return { id: x.assetIds[0] };
     });
 
+    // only the calls for CDF are cutomizable, not the once for geoSpatial
+    if (customFilter) {
+      return await customFilter(assetIds);
+    }
+
```

Comment:
When fetching wells by polygons, you still have to input a polygon as **wkt** or **GeoJson**, so that the geoSpatial method can get retrieve the asset ids. However, the you can customize the way Assets are being called in CDF and translated/wrapped to wells.

Reply:
Shouldn't the user still have the option of providing any method that receives a polygon (wkt or geojson) and returns a list of wells, regardless of how they find it (they may not use geospatial if they don't want to)

In truth, I think this whole method should be encapsulated in a higher level of abstraction.That is:
one method like this: 
```
public getByPolygon = async ({
    geometry,	   
    crs = 'epsg:4326',
    outputCrs = 'EPSG:4326',
    limit = 1000,
    offset = 0,
    customFilter = SearchWell,
  }
```
and, internally, this method would either return the custom filter (like the other ones you wrote) OR do what this current method is doing, calling geospatial, but defining the default arguments to find well heads in geospatial, like ` source = 'wellmodel, layer = 'point',	  attributes = ['geometry']`. 

All of those parameters would be fixed for us, and defined inside of the well data layer. If the user does not want to use geospatial, or wants to use it with their own custom version of these parameters, they provide the custom method.